### PR TITLE
salt: to 3001.1; python{,3}-pycryptodomex: new package

### PR DESCRIPTION
**Pull request by Goorzhel**
@Vaelatern 

With saltstack/salt#55310 wrapped up, I figured I'd take a stab at updating our `salt` package. This _should_ also obsolete saltstack/salt#56551.

```
[ag@server~]# salt-master -V
Salt Version:
           Salt: 3001.1

Dependency Versions:
           cffi: 1.14.0
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: 4.2.0
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
       M2Crypto: 0.35.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 1.0.0
   mysql-python: Not Installed
      pycparser: 2.19
       pycrypto: Not Installed
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.8.4 (default, Jul 17 2020, 11:12:18)
   python-gnupg: Not Installed
         PyYAML: 5.3
          PyZMQ: 19.0.2
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.2

System Versions:
           dist: void rolling void
         locale: utf-8
        machine: x86_64
        release: 5.7.10_1
         system: Linux
        version: void rolling void
```

### CODE REVIEW

**Comment by Goorzhel** [comment]
> srcpkgs/salt/template:7: do not set pycompile_module, it is autodetected
TIL about xlint. 😅

**Comment by Goorzhel** [comment]
Hmm:
```
ag@server $ sudo salt-key -L
Password:
Error: The 'pycryptodomex>=3.9.7' distribution was not found and is required by salt
ag@server $ xbps-query -f salt | grep requires | tee $(tty) | xargs cat
/usr/lib/python3.8/site-packages/salt-3001.1-py3.8.egg-info/requires.txt
Jinja2
msgpack>=0.5,!=0.5.5
PyYAML
MarkupSafe
requests>=1.0.0
distro
pycryptodomex>=3.9.7
pyzmq>=17.0.0
```
We have `pycryptodome` in void-packages, but not `pycryptodomex` (I imagine that's deliberate). For the time being I've papered over this with `sudo pip install pycryptodomex`.

**Comment by fosslinux** [comment]
I think the only reason that we don't have `pycryptodomex` is because no-one ever added it to the repositories. You should make a template for that and add it to this PR (I think), unless there is something I am missing about `pycryptodome` and `pycryptodomex`.

**Comment by ericonr** [comment]
I needed it for something once, and they are indeed different packages. I can't find the template for it, unfortunately.

**Comment by Goorzhel** [comment]
```
$ salt-master -V | grep cryptodome
   pycryptodome: Not Installed
$ xi python3-pycryptodomex
<...>
$ salt-master -V | grep cryptodome
   pycryptodome: 3.9.8
```
Not bad, not bad at all.

**Comment by fosslinux** [comment]
Correct commit naming format would be `New package: pycryptodomex-3.9.8`

**Comment by Goorzhel** [comment]
Whoops. Should I format the `salt` commit accordingly?

**Comment by fosslinux** [comment]
For salt, normally it would be `salt: update to 3001.1, (something else important)`, or the somethign else important on a new line what you have is currently fine for salt (although if you push again maybe s/upgrade/update`). New packages use the `New package: package-version` format though.

Here is the documentation on this https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#committing-your-changes

**Comment by Goorzhel** [comment]
I appreciate the pointers, @fosslinux — commits fixed.

**Comment by ericonr** [comment]
We aren't taking new python2 packages, so you can just build the python3 one without worrying about py2.

File: srcpkgs/salt/template [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-pycompile_module="salt"
-hostmakedepends="python-devel python-setuptools"
-depends="python-yaml python-Jinja2 python-requests python-pyzmq
- python-M2Crypto python-tornado python-msgpack dmidecode pciutils
- python-psutil"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3-yaml python3-Jinja2 python3-requests python3-pyzmq
+ python3-M2Crypto python3-tornado python3-msgpack dmidecode pciutils
+ python3-psutil python3-distro"
```
Comment (fosslinux) [inline srcpkgs/salt/template:10]: Should `python3-pycryptodomex` be added here?

Comment (Goorzhel) [reply srcpkgs/salt/template:10]: 🤦‍♂️ Dunno how I missed that.

**Comment by Vaelatern** [comment]
Thanks for the work on this, I'll review and get back on this.

**Comment by Vaelatern** [comment]
Looks like python3-pycrptodome already exists.

**Comment by Goorzhel** [comment]
It does, but Salt [seems rather particular](https://github.com/void-linux/void-packages/pull/24136#issuecomment-670962101) about having pycryptodomex as opposed to non-x. As far as I can tell, [their only difference](https://pycryptodome.readthedocs.io/en/latest/src/introduction.html?highlight=pycryptodomex) is that with-x installs to the Python module `Cryptodome`, and non-x installs to `Crypto`. @ericonr stated a need for pycryptodomex in the repo, anyway, so I figured I'd fill it.

_Hopefully_ this is my last force-push — I had to iron out some subtle typos.

**Comment by Vaelatern** [comment]
Thank you!

I'm going to test salt more, but I know this change does make our salt more usable than before!

**Comment by Goorzhel** [comment]
Glad to help!

**Pull request closed without merging**